source/bsd/kqueue_event_loop.c

```diff
@@ -366,10 +366,12 @@ static int s_run(struct aws_event_loop *event_loop) {
      * and it's ok to touch cross_thread_data without locking the mutex */
     impl->cross_thread_data.state = EVENT_THREAD_STATE_RUNNING;
 
+    aws_thread_increment_unjoined_count();
```

Comment:
Why isn't this managed by the thread wrapper itself? This seems really error prone, to force downstream code to effectively ref-count this, when the `aws_thread` code knows whether or not it created a thread.

Only an unmanaged thread should have to do this. And unmanaged threads should be EXTREMELY rare. In fact, I'm not sure when we'd want one.

Reply:
event loop threads aren't technically managed (there's a bunch of wrapping logic around their lifetime in the event loop platform impls) but I think it's best to have them counted as if they were.